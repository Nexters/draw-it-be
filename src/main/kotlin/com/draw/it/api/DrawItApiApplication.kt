package com.draw.it.api

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.scheduling.annotation.EnableScheduling

@SpringBootApplication
@ConfigurationPropertiesScan(basePackages = ["com.draw.it.api"])
@EnableScheduling
class DrawItApiApplication

fun main(args: Array<String>) {
	runApplication<DrawItApiApplication>(*args)
}
