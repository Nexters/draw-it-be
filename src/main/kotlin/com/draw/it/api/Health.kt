package com.draw.it.api

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.responses.ApiResponse
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@Tag(name = "Health", description = "헬스체크 API")
@RestController
class Health {

    @Operation(summary = "헬스체크", description = "서버 상태를 확인합니다")
    @ApiResponse(responseCode = "200", description = "서버 정상 상태")
    @GetMapping("/health")
    fun health() {
        return
    }
}