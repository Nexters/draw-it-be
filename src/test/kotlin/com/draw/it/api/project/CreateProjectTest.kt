package com.draw.it.api.project

import com.draw.it.common.IntegrationTest
import com.fasterxml.jackson.databind.ObjectMapper
import io.mockk.every
import io.mockk.mockkStatic
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.jdbc.Sql
import java.util.*

@IntegrationTest
@Sql("/project.sql")
class CreateProjectTest {
    @Autowired
    private lateinit var createProject: CreateProject

    @Autowired
    private lateinit var objectMapper: ObjectMapper

    @Test
    fun `새로운 프로젝트를 생성하고 저장한다`() {
        val request = CreateProjectRequest(
            topic = "테스트 프로젝트",
            message = "테스트 메시지",
            backgroundColor = "#FFFFFF",
        )
        // UUID.randomUUID().toString() should be mocked to return a consistent value for testing
        mockkStatic(UUID::class)
        every { UUID.randomUUID().toString() } returns "test-uuid"

        val response = createProject.createProject(1L, request)

        // JSON으로 변환 후 id 필드 제거하고 검증
        Approvals.verify(response.toJson())
    }

    private fun CreateProjectResponse.toJson(): String {
        return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(this)
    }
}