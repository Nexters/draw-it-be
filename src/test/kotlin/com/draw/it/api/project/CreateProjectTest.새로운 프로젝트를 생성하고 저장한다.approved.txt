{
  "id" : 1,
  "userId" : 1,
  "topic" : "테스트 프로젝트",
  "message" : "테스트 메시지",
  "backgroundColor" : "#FFFFFF",
  "uuid" : "test-uuid"
}